### PR TITLE
Add API to standalone to easily create an SvgLoader

### PR DESCRIPTION
**Pull request by rock3r**
Also:
* Show how to create icons in standalone sample
* Clean up related code

### CODE REVIEW

File: core/src/main/kotlin/org/jetbrains/jewel/Icon.kt [hunk 1]
```diff
@@ -90,22 +90,6 @@ fun Icon(
     )
 }
 
-@Composable
-fun Icon(
```
Comment (rock3r) [inline core/src/main/kotlin/org/jetbrains/jewel/Icon.kt:94]: Shouldn't be used — loads resources the wrong way. Hence it was removed.

File: core/src/main/kotlin/org/jetbrains/jewel/NoIndication.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package org.jetbrains.jewel
+
+import androidx.compose.foundation.Indication
+import androidx.compose.foundation.IndicationInstance
+import androidx.compose.foundation.interaction.InteractionSource
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.graphics.drawscope.ContentDrawScope
+
+object NoIndication : Indication {
```
Comment (rock3r) [inline core/src/main/kotlin/org/jetbrains/jewel/NoIndication.kt:9]: Moved from IntUI to core to be used in components as well

File: core/src/main/kotlin/org/jetbrains/jewel/Tabs.kt [hunk 3]
```diff
@@ -165,21 +161,6 @@ internal fun TabImpl(
     }
 }
 
-private object NoIndication : Indication {
```
Comment (rock3r) [inline core/src/main/kotlin/org/jetbrains/jewel/Tabs.kt:168]: Using the now-in-core NoIndication

File: int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/IntUiTheme.kt [hunk 4]
```diff
@@ -197,21 +197,40 @@ fun IntUiTheme(
...
+
+/**
+ * Create and remember an instance of [SvgLoader].
+ *
+ * Note that since [SvgLoader] may cache the loaded images, and
+ * that creating it may be somewhat expensive, you should only
+ * create it once at the top level, and pass it around.
+ */
+@Composable
+fun rememberSvgLoader(
```
Comment (rock3r) [inline int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/IntUiTheme.kt:218]: This the main addition, an API to create an `SvgLoader` as needed.

File: int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/IntUiTheme.kt [hunk 5]
```diff
@@ -197,21 +197,40 @@ fun IntUiTheme(
     swingCompatMode: Boolean = false,
     content: @Composable () -> Unit,
 ) {
-    val svgLoader by remember(themeDefinition.isDark, themeDefinition.iconData, themeDefinition.colorPalette) {
+    val svgLoader by rememberSvgLoader(
+        isDark = themeDefinition.isDark,
+        iconData = themeDefinition.iconData,
+        colorPalette = themeDefinition.colorPalette,
```
Comment (rock3r) [inline int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/IntUiTheme.kt:203]: Need to read these explicitly as the composition locals aren't set up yet here

File: int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/IntUiTheme.kt (hunk 4 above)
Comment (rock3r) [reply int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/IntUiTheme.kt:218]: Note: the bridge module has a different way to access the SvgLoader.

File: samples/standalone/src/main/kotlin/org/jetbrains/jewel/samples/standalone/components/Tabs.kt [hunk 6]
```diff
@@ -110,18 +130,50 @@ private fun EditorTabShowcase() {
...
+        LaunchedEffect(interactionSource) {
+            interactionSource.interactions.collect { interaction ->
+                when (interaction) {
+                    is HoverInteraction.Enter -> isHovered = true
+                    is HoverInteraction.Exit -> isHovered = false
+                }
+            }
+        }
+
+        // TODO create an IconButton instead of this hack
```
Comment (rock3r) [inline samples/standalone/src/main/kotlin/org/jetbrains/jewel/samples/standalone/components/Tabs.kt:154]: ...but that's for another day.

**Review by fscarponi — APPROVED** [review]

**Pull request merged**